### ISSUE TITLE
write_riemann: spurious messages

### ISSUE THREAD

**Issue by faxm0dem** [collectd/collectd#1234]
The following message has been annoying me for a while:

```
collectd write_riemann plugin: uc_get_rate failed.
```

It's not the message itself, but the frequency, which is just insane.
Any pointers on why these come up?

**Comment by AurelienLourot** [comment]
For us this happens consistently on 20+ macOS machines with collectd 5.7.1 if and only if collectd is built with the debug support and the `logfile/LogLevel` is set to `debug`.

**Comment by octo** [comment]
Related / possible duplicate: #1193

**Comment by octo** [comment]
@faxm0dem, @AurelienLourot: are all of these local values or some of them from a remote host?

**Comment by octo** [comment]
Sorry, I was confused. The roasted issue is actually #2209.

**Comment by octo** [comment]
@AurelienLourot since you have debugging enabled: do you see the following debug message pop up?

```
utils_cache: uc_get_rate_by_name: No such value: <name>
```

**Comment by octo** [comment]
There are x ways in which the `uc_get_rate()` function can fail:

1.  Creating the string for the lookup fails.
    → an **error** is printed
2.  The number of values expected and returned doesn't match.
    → an **error** is printed
3.  Allocating memory for the values fails.
    → an **error** is printed
4.  The metric is *not* found in the cache.
    → a **debug** message is printed
5.  The metric *is* found in the cache, but marked *MISSING*.
    → no message is produced.

Based on your description, I'd rule out 1, 2 and 3. Number 5 is the most likely candidate for me. Do you have the chance to build *collectd* from source and try something for me? I can't reproduce the problem with just the CPU and CSV plugins …

**Comment by faxm0dem** [comment]
I don't see the message anymore, but I'm running 5.6.2 on CentOS

**Comment by AurelienLourot** [comment]
@octo It's actually number 4, always, i.e.
```
[2017-05-19 00:04:33] [debug] utils_cache: uc_get_rate_by_name: No such value: plsrvmp-qbs27/swap/swap-free
[2017-05-19 00:04:33] [error] write_riemann plugin: uc_get_rate failed.
```

**Comment by octo** [comment]
@AurelienLourot which version of collectd are you running?

**Comment by AurelienLourot** [comment]
5.7.1, see https://github.com/collectd/collectd/issues/1234#issuecomment-302557803

**Comment by octo** [comment]
So it looks like metrics are purged from the cache even though they're still "active". #1193 is a likely candidate for this. That bug was introduced in 5.7.0 and was fixed with 5.7.2.

@AurelienLourot do you still observe this bug with 5.7.2?

**Comment by AurelienLourot** [comment]
@octo Indeed, this is fixed in 5.7.2, thanks!

**Comment by octo** [comment]
Thanks for the feedback @AurelienLourot!
